lib/client/api.go

```diff
@@ -584,15 +587,10 @@ func RetryWithRelogin(ctx context.Context, tc *TeleportClient, fn func() error,
 	}
 	log.Debugf("Activating relogin on %v.", fnErr)
 
-	// check if the error is a private key policy error.
-	if privateKeyPolicy, err := keys.ParsePrivateKeyPolicyError(fnErr); err == nil {
-		// The current private key was rejected due to an unmet key policy requirement.
-		fmt.Fprintf(tc.Stderr, "Unmet private key policy %q\n", privateKeyPolicy)
-		fmt.Fprintf(tc.Stderr, "Relogging in with hardware-backed private key.\n")
-
-		// The current private key was rejected due to an unmet key policy requirement.
-		// Set the private key policy to the expected value and re-login.
-		tc.PrivateKeyPolicy = privateKeyPolicy
+	if privateKeyPolicy, parseErr := keys.ParsePrivateKeyPolicyError(fnErr); parseErr == nil {
```

Comment:
I misunderstood the call to this function two times in different ways and I wonder how we could make it more clear what happens here.

To me it's not obvious that it's okay to ignore `parseErr`. What if this call would look something like this?

```suggestion
	if privateKeyPolicy, ok := keys.IsPrivateKeyPolicyError(fnErr); ok {
```

Reply:
Fixed in e41df613d76825777b3e63dcc4f0fdb100229657